File: Cpp/693_Binary_Number_with_Alternating_Bits.cpp
class Solution {
public:
    bool hasAlternatingBits(int n) {
        int previous = n%2;
        int i = n/2;
        while(i>0){
            int remainder = i%2;
            if(previous==remainder) return false;
            previous = remainder;
            i=i/2;
        }
        return true;
    }
};

File: Kotlin/693_Binary_Number_with_Alternating_Bits.kt
class Solution {
    fun hasAlternatingBits(n: Int): Boolean {
        var previous = n % 2
        var i = n / 2
        while(i>0){
            val remainder = i % 2
            if(previous == remainder) return false
            previous = remainder
            i = i / 2
        }
        
        return true
    }
}